tensorflow/compiler/xla/service/gpu/cusolver_context.cc

```diff
@@ -35,12 +52,25 @@ template <>
 struct CUDAComplexT<std::complex<double>> {
   typedef cuDoubleComplex type;
 };
+#else
+// can't use gpuFloatComplex, gpuDoubleComplex, because e.g.
+// hipFloatComplex and rocblas_float_complex are two unrelated types
+template <>
+struct CUDAComplexT<std::complex<float>> {
+  typedef rocblas_float_complex type;
+};
+template <>
+struct CUDAComplexT<std::complex<double>> {
```

Comment:
Same comment about preferring platform-agnostic names? Do you think it would be too hard to do?